tests/wake-format/basic/stdout

```diff
@@ -341,6 +329,86 @@ def ff =
     .ggggggggggg
     .hhhhhhhhhhh
 
+def ff =
+    aaaaaaaaaaaaa +
+    bbbbbbbbbbbbbbbb +
+    cccccccccccccccccc +
+    ddddddddddddddddddd +
+    eeeeeeeeeeeeeee +
+    ffffffffff +
+    ggggggggggg +
+    hhhhhhhhhhh
+
+def ff =
+    aaaaaaaaaaaaa *
+    bbbbbbbbbbbbbbbb *
+    cccccccccccccccccc *
+    ddddddddddddddddddd *
+    eeeeeeeeeeeeeee *
+    ffffffffff *
+    ggggggggggg *
+    hhhhhhhhhhh
+
+def ff =
+    aaaaaaaaaaaaa ||
+    bbbbbbbbbbbbbbbb ||
+    cccccccccccccccccc ||
+    ddddddddddddddddddd ||
+    eeeeeeeeeeeeeee ||
+    ffffffffff ||
+    ggggggggggg ||
+    hhhhhhhhhhh
+
+def ff =
+    aaaaaaaaaaaaa $$
+    bbbbbbbbbbbbbbbb $$
+    cccccccccccccccccc $$
```

Comment:
Also, it doesn't space afterward (`..aaaaaaaaaaaaa`) which might be another thing to think about -- would we rather `a .. b`?

Reply:
AFAICT, `..` isn't a used operator (at the very least its very uncommon if it is). The reason there are exceptions for `|` and `$` is that `||` is a very common operator and `$` being the cousin of `|` we except `$$` for consistency.

I'd personally not want to make exceptions for `..` until it becomes necessary. Adding exceptions for the same TOKEN class is a bit of a pain and I'd like to only do it when needed

Reply:
I definitely wasn't suggesting that we want proactive exceptions for yet-unused specific operators.  That was an observation that `|` is prefix but `||` is suffix so presumably the specific operator is the scope for the exception in that case; however, dropping down a few tests in the newly-added stdout, [the block for `..`](https://github.com/sifive/wake/blob/91168e8ce0bcb971ec5e0a2cca4a4c1f08a5d5ac/tests/wake-format/basic/stdout#L382-L390) currently has that operator prefixed when I would have expected it be suffixed like all the other double-character operators you added tests for, so presumably the scope of the exception for `.` is for its *class* rather than just the one operator.  The better illustration would be that if I were to try to reimplement Aeson's default-fallback operator for the (old) Wake JSON library, I'd likely wind up with such spacing as `json // field.!=default` rather than the expected `json // field .!= default` -- in other words, I think the current special case logic is too broad and should be restricted.

Like you say, though, `.` seems to be the *only* dot-class operator currently defined (searched public and internally for `def [^=]*[ )]\.`), so it's not something we have to worry about for the MVP.  Restricting that is definitely something I'd like to see eventually rather than having someone stumble across it after we've moved on to other areas of the code, but it's not a blocker for the moment.

Reply:
Ah right. Yeah that is a good point. There is some weirdness there and the exception for `.` is a bit more broad. Happy to review the rule there post MVP :)

One quick question re: `.!=` from Aeson. How would you expect `json // field .!= default` to bind?  `(json // field) .!= default` or `json // (field .!= default)`?

Reply:
Right I assumed as much :) The `.!=` operator wouldn't work there because of how strongly _all_ `.` ops bind. In the case of `json // field .!= default`, I'd personally want it to format as `json // field.!=default` to call out that strong binding.

I'm still willing to hear the argument about prefix/suffix for `.<stuff>`  but it should pretty much always not space imo

Reply:
Oh, right, binding precedence is also part of the class; I should have remembered that.  Yeah, that's a good reason to not space.